processing/src/main/java/org/apache/druid/query/QueryMetrics.java

```diff
@@ -242,9 +243,34 @@ public interface QueryMetrics<QueryType extends Query<?>>
 
   void segment(String segmentIdentifier);
 
-  void preFilters(List<Filter> preFilters);
+  /**
+   * @deprecated use {@link #filterBundle(FilterBundle.BundleInfo)} instead to collect details about filters which were
+   * used to construct {@link org.apache.druid.segment.BitmapOffset} or
+   * {@link org.apache.druid.segment.vector.BitmapVectorOffset}
+   */
+  @Deprecated
+  @SuppressWarnings({"unreachable", "unused"})
+  default void preFilters(List<Filter> preFilters)
+  {
+    // do nothing, nothing calls this
```

Comment:
yea, since you currently have to write an extension to customize metrics, i figured it would be less disruptive to deprecate than remove it immediately. Will update the javadocs to indicate that it will go away in the future.